### ISSUE TITLE
Rest API: calling API in given groupId context

### ISSUE THREAD

**Issue by piotreke12** [fossology/fossology#1529]
### Description

Im not sure if it is a bug or just a behaviour, so I report this as feature request

# Intro
Groups in Fossology work rather like context than standard group of users.
In the GUI, User has to change Group context (upper right corner of page) to do things for uploads he has access because of being in specific group, while "standard" group behaviour should always show and allow user to do things with all uploads he has access to within all his groups.

# Problem

Because of this Group behavior, there is a problem when calling REST API.
In general, REST API calls uses (stores session parameters) always with GroupID set as "default" in *users* table (group_fk column).
The thing is, this groupId stored in database changes each time when given user changes "selectMemberGroup" item on GUI

Few weeks ago, @GMishx solved #1468 issue, and added option to store new upload for specific groupId given as REST call parameter.
It works ok, but there is a problem when:
- user has 2 groups access: group A and B
- user on GUI selected group **B** (upper-right select box), so this value is stored in DB
- user calls REST API (with his own token) and creates upload sending groupID pointing to group **A** (he has access, so he can) and this upload ic created in group **A** context
- user tries to run jobs via REST API (his own token) and he **can not** (no access error) - because /jobs endpoint uses groupId stored in database, which points to group **B** selected in GUI 

In my oppinion, there should be option for all REST API calls to pass specific groupId for which you want to do stuff 

In fact, I did some research, and if this explanation is clear for You and you also think there is a bug to solve, I have a proposal :)

## Resolve proposal

In RestAuthMiddleware class, there is a method `$authHelper->updateUserSession` call that sets current Session parameters. This method uses `$row = $this->userDao->getUserAndDefaultGroupByUserName($user["name"]);` call that reads user default groupId from DB.

In my oppinion, we should change this method and when someone sends groupId in request, we should update session with groupId from request (if there is no groupId provided, we could still use this value from DB)

I think this change would also be valid for /upload endpoint, so changes done by @GMishx in #1498 would be no longer needed - but im not 100% sure.

**To be honest, i made some work on my side to resolve this issue as described above (as we are going to use this REST API for our CI/CD intergation and we need this to work correctly ASAP) so I think I can prepare PullRequest for your verification if you accept this way of doing it**

By the way, I think it would be nice to add also groupName parameter, as GroupId is quite hard to find for standard GUI user

**Comment by GMishx** [comment]
Hello @piotreke12,
This is much more cleaner and better approach than what I did in #1498 and it can be reverted once your pull request is ready.

Also, it is a nice approach to use `groupName` instead of `groupId`. I was also thinking to provide the groups which user is part of as a result of `/users/userid` if requestor is an admin or querying own user.
